tests/unit/python/foglamp/services/south/test_ingest.py

```diff
@@ -0,0 +1,588 @@
+storage = None  # type: Storage
+_readings_stats = 0  # type: int
+_discarded_readings_stats = 0  # type: int
+_sensor_stats = {}  # type: dict
+_write_statistics_task = None  # type: asyncio.Task
+_write_statistics_sleep_task = None  # type: asyncio.Task
+_stop = False
+_started = False
+_readings_lists = None  # type: List
+_current_readings_list_index = 0
+_insert_readings_tasks = None  # type: List[asyncio.Task]
+_readings_list_batch_size_reached = None  # type: List[asyncio.Event]
+_readings_list_not_empty = None  # type: List[asyncio.Event]
+_readings_lists_not_full = None  # type: asyncio.Event
+_insert_readings_wait_tasks = None  # type: List[asyncio.Task]
+_last_insert_time = 0  # type: int
+_readings_list_size = 0  # type: int
+_write_statistics_frequency_seconds = 5
+_readings_buffer_size = 500
+_max_concurrent_readings_inserts = 5
+_readings_insert_batch_size = 100
+_readings_insert_batch_timeout_seconds = 1
+_max_readings_insert_batch_connection_idle_seconds = 60
+_max_readings_insert_batch_reconnect_wait_seconds = 10
+category = 'South'
+default_config = {
+    "write_statistics_frequency_seconds": {
+        "description": "The number of seconds to wait before writing readings-related "
+                       "statistics to storage",
+        "type": "integer",
+        "default": str(_write_statistics_frequency_seconds)
+    },
+    "readings_buffer_size": {
+        "description": "The maximum number of readings to buffer in memory",
+        "type": "integer",
+        "default": str(_readings_buffer_size)
+    },
+    "max_concurrent_readings_inserts": {
+        "description": "The maximum number of concurrent processes that send batches of "
+                       "readings to storage",
+        "type": "integer",
+        "default": str(_max_concurrent_readings_inserts)
+    },
+    "readings_insert_batch_size": {
+        "description": "The maximum number of readings in a batch of inserts",
+        "type": "integer",
+        "default": str(_readings_insert_batch_size)
+    },
+    "readings_insert_batch_timeout_seconds": {
+        "description": "The number of seconds to wait for a readings list to reach the "
+                       "minimum batch size",
+        "type": "integer",
+        "default": str(_readings_insert_batch_timeout_seconds)
+    },
+    "max_readings_insert_batch_connection_idle_seconds": {
+        "description": "Close storage connections used to insert readings when idle for "
+                       "this number of seconds",
+        "type": "integer",
+        "default": str(_max_readings_insert_batch_connection_idle_seconds)
+    },
+    "max_readings_insert_batch_reconnect_wait_seconds": {
+        "description": "The maximum number of seconds to wait before reconnecting to "
+                       "storage when inserting readings",
+        "type": "integer",
+        "default": str(_max_readings_insert_batch_reconnect_wait_seconds)
+    },
+}
+
```

Comment:
@amarendra-dianomic not sure why do we need a copy paste of all above from original code file. should not we use these as imported var /def ?

Reply:
done via another way.